rclrs_examples/src/rclrs_subscriber.rs

```diff
@@ -18,5 +19,9 @@ fn main() -> rclrs::RclResult {
         },
     )?;
 
-    rclrs::spin(&node)
+    // rclrs::spin(&node).with_context(|| "RCL Error Occurred")
+    match rclrs::spin(&node) {
+        Ok(()) => Ok(()),
+        Err(RclError) => Err(anyhow::Error::from(RclError)),
```

Comment:
Is this correct? I don't see RclError declared anywhere.

Reply:
Ah, I see, thanks.

In that case, I'd get rid of the RCLError struct and rename RCLStatusCode to RclError, there's a naming inconsistency here RCL vs Rcl, I don't mind either way, as long as it's consistent. I wonder if there's a way to automatically generate RclError from the status codes in rcl.h, might be worth exploring that.

Reply:
Oh, good catch on the inconsistency...

I was thinking of renaming RCLStatusCode to RclError, but one of the status codes actually pertains to a non-error: "Ok". I haven't figured out a nice way to deal with that, since it feels very confusing to have a non-error "Error". If you know what I mean?

And the current way to generate `RclStatusCode`s is that the enum implements the `From<i32>` trait. Though you have to be careful with that - if you pass an unsupported enum, you get an `unimplemented!` panic. See `rclrs_common/src/lib.rs`, line 76 for the current implementation.

Reply:
> I haven't figured out a nice way to deal with that, since it feels very confusing to have a non-error "Error". If you know what I mean?

Yeah, I get you, I think that's why I added an `RclResult` struct, but I can't remember for sure. In any case I like how succinct your solution is. Maybe `RclStatus`? Though it does not convey when it's an actual error. Anyway, let's keep it as `RclError`.

I thought of generating `RclError` using `libclang` and `bindgen`, but on a second thought, doesn't make much sense because the error values won't change often.

Reply:
I think what I'll do is leave this for a general error streamlining process. I'm thinking that perhaps the best way to handle this is to wrap all the `unsafe` `rcl` calls and convert their outputs to things that Rust is more familiar with handling (i.e: Result types). This could be doable with a quick tweak in the `rclrs_common/src/library.rs` code, where instead of a `From<i32> for RclError` we could figure out a function to convert said `i32` into a full-fledged Result type.

But that's for another PR, if it's possible.